src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -0,0 +1,151 @@
+      "the", "is", "for", "in", "of", "so", "to"
+      ));
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    //get location
+    //filter by location and cutoff outside it
+    
+    //get tags
+    //drop all without first tag
+    //those with most tags in common with search go to top
+    //those closest to the user go to the top
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+  }
+
+  /**
+   * Returns keywords from an event (currently using just the title
+   * and description) based off an algorithm
+   */
+  public static List<String> getKeywords(String title, String desc) {
+    //TODO: convert to lowercase in processing (figure out acronyms)
+    List<String> titleWords = SearchServlet.getSeparateWords(title);
+    
+    //process title words
+    Map<String, Integer> titleMap = new HashMap<String, Integer>();
+    for (String w : titleWords) {
+      w = w.toLowerCase();
+      if (titleMap.containsKey(w)) {
+        titleMap.put(w, titleMap.get(w) + 1);
+      } else {
+        titleMap.put(w, 1);
+      }
+    }
+    List<String> descWords = SearchServlet.getSeparateWords(desc);
+    //process desc words
+    Map<String, Integer> descMap = new HashMap<String, Integer>();
+    for (String w : descWords) {
+      w = w.toLowerCase();
+      if (descMap.containsKey(w)) {
+        descMap.put(w, descMap.get(w) + 1);
+      } else {
+        descMap.put(w, 1);
+      }
+    }
+
+    List<Map.Entry<String, Integer>> titleMapList = new ArrayList<Map.Entry<String, Integer>>(titleMap.entrySet()); 
+
+    //merge lists
+    List<Map.Entry<String, Integer>> mergeList = new ArrayList<Map.Entry<String, Integer>>(descMap.entrySet());
+    for (Map.Entry titleEntry : titleMapList) {
+      boolean found = false;
+      for (int i = 0; i < mergeList.size(); i++) {
+        Map.Entry mergeEntry = mergeList.get(i);
+        if (mergeEntry.getKey().toString().equals(titleEntry.getKey().toString())) {
+          mergeEntry.setValue(((int) mergeEntry.getValue()) + ((int) titleEntry.getValue() * 2));
+          found = true;
+        }
+      }
+      if (!found) {
+        Map.Entry<String,Integer> entry = 
+            new AbstractMap.SimpleEntry<String, Integer>(
+            titleEntry.getKey().toString(), ((int) titleEntry.getValue()) * 2 );
+        mergeList.add(entry);
+      }
+    }
+
+    Collections.sort(mergeList, new Comparator<Map.Entry<String, Integer>>() { 
+      public int compare(Map.Entry<String, Integer> o1,  
+                         Map.Entry<String, Integer> o2) { 
+        return (o2.getValue()).compareTo(o1.getValue()); 
+      } 
+    });
+
+    //add top results to the final list
+    List<String> finalList = new ArrayList<String>();
+    int count = 0;
+    //minimum number of instances to be considered a keyword
+    int cutoff = 2;
+    while (finalList.size() < 5) {
```

Comment:
It would be good to store these parameters as class-level named constants with comments, for example:

// Number of keywords to produce.
public static final int NUM_KEYWORDS = 5;

Reply:
Got it!